openstack_dashboard/dashboards/project/instances/tables.py

```diff
@@ -1308,3 +1310,18 @@ class Meta(object):
                        EditInstance, ConsoleLink, SoftRebootInstance,
                        RebootInstance, StopInstance, RebuildInstance,
                        DeleteInstance)
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        user_id_column = next((c for c in self.get_columns() if c.name == "user_id"), None)
+        if user_id_column:
+            user_id_column.filters.append(lambda u: self.uid_to_user(u))
+
+    def uid_to_user(self, uid):
+        if not uid:
+            return None
+        try:
+            user = api.keystone.user_get(self.request, uid, admin=False)
```

Comment:
Is it possible to batch these at all? I assume not built into keystone...

Reply:
Added a cache like [this commit](https://github.com/ChameleonCloud/blazar-dashboard/pull/35/commits/3134d51e95a5bc551eae713969599f1d8371718d) from blazar-dashboard.